lib/fake_async.dart

```diff
@@ -232,7 +233,7 @@ class FakeAsync {
     for (;;) {
       if (_timers.isEmpty) break;
 
-      var timer = minBy(_timers, (timer) => timer._nextCall);
+      var timer = minBy(_timers, (FakeTimer timer) => timer._nextCall)!;
```

Comment:
What caused this? Does minBy always return nullable? Why do we need the argument type specified in the callback?

Reply:
This is just for type inference purposes, it was inferring `Object?` before I added this.

I could explicitly add the generics but there are actually two so this is less verbose.

Reply:
Also yes `minBy` returns nullable, because it returns null if the list is empty. This is one of the cases where we would possibly make a breaking change if we could, and throw an ArgumentError in that case instead.